src/state.rs

```diff
@@ -272,12 +290,16 @@ impl<'a, T, E: Send + Sync + 'static> StateMachine<'a, T, E> {
     ///
     /// # Panics
```

Comment:
You forgot to update the documentation. It doesn't panic anymore but returns the StateError